### PR TITLE
Fix Issue 5215 - HashDisclosureScanner ignore obvious jsessionid Values

### PR DESCRIPTION
**Pull request by kingthorin**
- HashDisclosureScanner - Update MD4/MD5 regex. https://www.regexplanet.com/share/index.html?share=yyyyu0ua0ar
- HashDisclosureScannerUnitTest - Created, adding unit tests.
- ZapAddOn.xml - Updated changes section with new entry.

Fixes zaproxy/zaproxy#5215

### CODE REVIEW

File: test/org/zaproxy/zap/extension/pscanrulesAlpha/HashDisclosureScannerUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        return new HashDisclosureScanner();
+    }
+
+    @Before
+    public void before() throws HttpMalformedHeaderException {
+        rule.setAlertThreshold(AlertThreshold.LOW); // Required by MD4/MD5 tests
+    }
+
+    @Test
+    public void shouldRaiseAlertWhenResponseContainsUpperHash() throws Exception {
```
Comment (thc202) [inline test/org/zaproxy/zap/extension/pscanrulesAlpha/HashDisclosureScannerUnitTest.java]: The names should be explicit about the hash being tested (it has many).

Comment (kingthorin) [reply test/org/zaproxy/zap/extension/pscanrulesAlpha/HashDisclosureScannerUnitTest.java]: Yup that makes sense. Will fix up.

**Comment by kingthorin** [comment]
Unittest naming change force pushed.

**Comment by thc202** [comment]
Thank you!

**Pull request merged**